caracal/workers/prep_worker.py

```diff
@@ -2,29 +2,124 @@
 import os
 import sys
 import caracal
+from caracal.dispatch_crew.utils import closeby
+import caracal.dispatch_crew.caltables as mkct
 import numpy as np
 from caracal.workers.utils import manage_flagsets as manflags
 from caracal.dispatch_crew import utils
+from astropy.coordinates import SkyCoord
 
 NAME = "Prepare Data for Processing"
 LABEL = 'prep'
 
+def getfield_coords(info, field, db, tol=2.9E-3, tol_diff=4.8481E-6):
+    """
+    Shameless copy of the hetfield function to return field coordinates 
+    from the database.
+    Find match of fields in info
+    Parameters:
+    info (dict): dictionary of obsinfo as read by yaml
+    field (str): field name
+    db (dict):   calibrator data base as returned by
+                 calibrator_database()
+    Go through all calibrators in db and return the first that matches
+    the coordinates of field in msinfo. Return empty string if not
+    found.
+    If coordinates difference is larger than tol_diff, return the correct coordinates, else return empty string.
+    """
+
+    # Get position of field in msinfo
+    ind = info['FIELD']['NAME'].index(field)
+    firade = info['FIELD']['DELAY_DIR'][ind][0]
+    firade[0] = np.mod(firade[0],2*np.pi)
+
+    dbcp = db.db
+    for key in dbcp.keys():
+        carade = [dbcp[key]['ra'],dbcp[key]['decl']]
+        if closeby(carade, firade, tol=tol):
+            if not closeby(carade, firade, tol=tol_diff):
+               return key, dbcp[key]['ra'], dbcp[key]['decl']
+            else :
+               return key, '', ''
+    return False
+
+
 def worker(pipeline, recipe, config):
     label = config['label_in']
     wname = pipeline.CURRENT_WORKER
     field_name = config["field"]
-
+     
     msdir = pipeline.msdir
     for i in range(pipeline.nobs):
         prefix_msbase = pipeline.prefix_msbases[i]
         mslist  = pipeline.get_mslist(i, label, target=(field_name == "target"))
 
+
         for msname in mslist:
             if not os.path.exists(os.path.join(msdir, msname)):
                 caracal.log.error(f"MS file {msdir}/{msname} does not exist. Please check that is where it should be.")
                 raise IOError
-
+            
+            if pipeline.enable_task(config, 'fixcalcoords'):
+               tol = config["fixcalcoords"]["tol"]
+               tol_diff = config["fixcalcoords"]["tol_diff"]
+               db = mkct.calibrator_database()
+               dbc = mkct.casa_calibrator_database()
+               msdict = pipeline.get_msinfo(msname)
+               for f in pipeline.bpcal[i]:
+                   fielddb, ra_corr, dec_corr = getfield_coords(msdict, f, db, tol = tol, tol_diff = tol_diff)
+                   if ra_corr:
+                      caracal.log.info("The calibrator coordinates in the measurement set are incorrect.")
```

Comment:
Can we add the name of the source to the log message, and make it a bit less cryptic overall? Perhaps: "The coordinates of calibrator {f} in the MS are offset. This is a known problem for some vintage MeerKAT MSs. We will now attempt to fix this by rephasing the visibilities using the CASA fixvis task."

Reply:
Done.